### ISSUE TITLE
Project: electrifier integrated desktop environment

### ISSUE THREAD

**Issue by tajbender** [microsoft/hack-together#188]
### Project name

electrifier

### Description

# about electrifier
## feature set
- [x] Manage your files. Locally. In the cloud. In your digital history.
- [ ] Project management.
- [ ] Built in viewers.
- [ ] Built in editors.
- [ ] Customize your own desktop environment.
- [ ] Synchronize workbenches around users, sessions and devices.
- [ ] [Graph](https://developer.microsoft.com/de-de/graph/) for Cloud Features

## contributions
- [@dahall/Vanara](https://github.com/dahall/Vanara) by David Hall @dahall.
Not that other WinAPI-Abstraction Layer but a complete managed shell around _about every native windows API_.
Also a great source for good practices. And very helpful community member 🥇 

- [@harborsiem/WindowsRibbon](https://github.com/harborsiem/WindowsRibbon) by @harborsiem
c# Wrapper around the native Windows Ribbon control.
the best ribbon for [WinForms](https://github.com/dotnet/winforms)

## ß test:
- Tobias

## download 
_electrifier is still in development. No stable builds available yet._

### current beta builds
*Attention*: Beta builds aren't supposed to be used in productive manner.

[v1.23.3.1 - *unsigned* *beta* - Imperial March Edition](https://github.com/electrifier/electrifier/releases/tag/v1.23.3.1)

### old version (using .net framework.)
The original .net-version of electrifier just brought tab-support to Windows File Explorer. Currently suspended.

See: https://github.com/electrifier/winforms.electrifier.org

## screenshots
![electrifier - file manager view](https://user-images.githubusercontent.com/10961499/225398092-6fee22df-9885-4d16-9bf7-c133e23d282d.jpg)

https://github.com/electrifier/electrifier/wiki/Screenshots-of-electrifier

### Repo URL

https://github.com/electrifier/electrifier

### Team members

tajbender, harborsiem, dahall

**Comment by tajbender** [comment]
[Hack-Together - Introduction: electrifier](https://github.com/microsoft/hack-together/issues/38)

**Comment by waldekmastykarz** [comment]
Cool @tajbender! Would you be willing to record a short video to demonstrate the app? It's not required for the hackathon but it would help us showcase your work internally at Microsoft and publicly in the community 😊

**Comment by tajbender** [comment]
> Cool @tajbender! Would you be willing to record a short video to demonstrate the app? It's not required for the hackathon but it would help us showcase your work internally at Microsoft and publicly in the community 😊

Good morning, @waldekmastykarz:

Yes, sir, I'll do my best - still a li'l tired cause of the hackathon. Need more lifeblood first. ☕ ⚰️ 

😁

**Comment by tajbender** [comment]
> Good morning, @waldekmastykarz:
> 
> Yes, sir, I'll do my best - still a li'l tired cause of the hackathon. Need more lifeblood first. ☕ ⚰️
> 
> 😁

Oh, and sorry for the duplicate issue. Just wanted to make sure we're in 🧷

**Comment by waldekmastykarz** [comment]
Hey @tajbender, it seems like we're missing registration information for your team mates. Could you please ask them to register for the hackathon asap to complete your submission?

**Comment by tajbender** [comment]
@waldekmastykarz
for your convience, the link to youtube. No sound, don't have a mic 😬 

https://youtu.be/ZbIb03M76Hg

**Comment by aycabas** [comment]
Hi @tajbender, we still can't find your teammates in the registrations. Can you please make sure that they are registered within today to be eligible for the prizes:  https://aka.ms/hack-together/register

**Comment by tajbender** [comment]
Hi, Ayca Bas,

tajbender here. Regarding my project, electrifier/electrifier: integrated desktop environment (github.com)<https://github.com/electrifier/electrifier>:
[https://opengraph.githubassets.com/724d4f162f01770f0fb7582980060e4e85b7ec5f7e3d540da65c345f8f2d3354/electrifier/electrifier]<https://github.com/electrifier/electrifier>
GitHub - electrifier/electrifier: integrated desktop environment<https://github.com/electrifier/electrifier>
integrated desktop environment. Contribute to electrifier/electrifier development by creating an account on GitHub.
github.com

I wasn't able to reach the original authors of https://github.com/harborsiem/WindowsRibbon
[https://opengraph.githubassets.com/5b392d79240b4e2068f39308de34fc6d33a3a87546a54cd1d94e3cca0d4d68ab/harborsiem/WindowsRibbon]<https://github.com/harborsiem/WindowsRibbon>
GitHub - harborsiem/WindowsRibbon: Microsoft Ribbon for .NET (Wrapper for C# and VB)<https://github.com/harborsiem/WindowsRibbon>
Microsoft Ribbon for .NET (Wrapper for C# and VB). Contribute to harborsiem/WindowsRibbon development by creating an account on GitHub.
github.com


..and https://github.com/dahall/Vanara
[https://opengraph.githubassets.com/f4ac70ca04cc3c918f5fc17b5c9ae3e9148a018c47cd56c35720b33b5042dbfd/dahall/Vanara]<https://github.com/dahall/Vanara>
GitHub - dahall/Vanara: A set of .NET libraries for Windows implementing PInvoke calls to many native Windows APIs with supporting wrappers.<https://github.com/dahall/Vanara>
A set of .NET libraries for Windows implementing PInvoke calls to many native Windows APIs with supporting wrappers. - GitHub - dahall/Vanara: A set of .NET libraries for Windows implementing PInvo...
github.com

so far.

However, for the version I was able to finish for Hack-Together, these components weren't in use anyhow.

The only person I could mention for the version of electrifier (https://github.com/electrifier/graph.electrifier/releases/tag/Code-Freeze-Hackathon) beside myself is my nephew, Tobias Jung. However, he has no GitHub-Account. He was only testing the Setup for Windows 10... But he has an XBLive-Account, "Tobin4tor". Anyways, I don't know If he is eligalbyle to be mentioned as Author for this.

So beside him as a test-dummy, I don't have any other teammates that were involved in that particular version.

Sorry for confusion,
regards,

tajbender aka Thorsten Jung @ electrifier



________________________________
Von: Ayca Bas ***@***.***>
Gesendet: Montag, 20. März 2023 09:31
An: microsoft/hack-together ***@***.***>
Cc: Thorsten Jung ***@***.***>; Mention ***@***.***>
Betreff: Re: [microsoft/hack-together] Project: electrifier integrated desktop environment (Issue #188)


Hi @tajbender<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Ftajbender&data=05%7C01%7C%7Ce1e8c5c809b04252ac5908db291d8663%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638148978995955428%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=XYCHWo5p7q%2F6gd4K2y8znVWI7JTiRlVKp%2FP49UXGK%2Fs%3D&reserved=0>, we still can't find your teammates in the registrations. Can you please make sure that they are registered within today to be eligible for the prizes: https://aka.ms/hack-together/register<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Faka.ms%2Fhack-together%2Fregister&data=05%7C01%7C%7Ce1e8c5c809b04252ac5908db291d8663%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638148978996111639%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WSBonSj2WIVt3hPzQH76JM8x2Z9JdZ3rc%2F30KEs0%2BZ0%3D&reserved=0>

**Comment by waldekmastykarz** [comment]
Hey @tajbender, we're reviewing you're hack and noticed that it seems like you're not using the Microsoft Graph .NET SDK which is a prerequisite for this hackathon. We see this one reference to the SDK in your code https://github.com/waldekmastykarz/ht23-electrifier/blob/f308886a731bd2a8a95ed181790abd09e8b4a83f/src/electrifier/Views/WorkbenchPage.xaml.cs, but it doesn't seem like it's doing anything. Are we missing something?

**Comment by tajbender** [comment]
Hi @waldekmastykarz,

well, no, that particular version I managed to get done in time does not more than logging in on the start page without further functionality and using Microsoft 365 inside the WebView2 😞.

That's because I ran intro build/deployment issues: Finally, time ran out, so for the deadline this is all that I managed to get done, beside the code fragments that have been commented out in the tagged version for the contest to track down the issues I had. These are the first artifacts of the Graph-Client I was working on till the last minutes of the deadline 🏃‍♂️. 

If that's not enough, then I just had bad luck. You don't have to worry about, I didn't expect to win a price for that messed and unfinished version. I just contributed it to qualify for the badge.

But as said, this is not what I hopped to get done for the contest...

However, the contest was a lot of fun and I learned tons of new things. Thanks for your time, efforts, and patience during the contest.

We'll meet again anytime soon I guess 😀,
regards,
tajbender
